### PR TITLE
reworked docker dev, added docker-compose builder

### PR DESCRIPTION
**Pull request by primeapple**
A first idea on #699 
@nukeop , please check if you like the changes. Also please check if you are able to bring docker up. I have some troubles with XHOST and get stuck when trying to launch the app with `docker-compose up`.

The instructions how to run the docker are in the README changes.

Inspiration is from https://hackernoon.com/a-better-way-to-develop-node-js-with-docker-cd29d3a0093

### CODE REVIEW

File: docker-compose.builder.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+services:
+  install:
+    image: node:14
+    volumes:
+      - .:/nuclear/
+    working_dir: /nuclear/
+    command: bash -xc "
+        npm install;
+        npx lerna bootstrap"
```
Comment (nukeop) [inline docker-compose.builder.yml:12]: `lerna bootstrap` will run as a postinstall step, so all we need is `npm install`.

Btw, I read that article some time ago, and again now, and I still don't see what we gain by splitting this part into a separate file, as opposed to a service in the same docker-compose file as the rest.

**Comment by nukeop** [comment]
Hey, I finally had some time to take a look at this. I made one comment; otherwise it's fine.

**Comment by primeapple** [comment]
Hey, thank you so much! :-)

After working and trying on docker some more, I am not so happy for this... Docker seems awesome for production, but for development... Well, I am not so sure about this.
I will look into it a bit more, but maybe it is the best, to just leave it as it is in master currently and flag it as "experimental" in the readme?

**Comment by nukeop** [comment]
I'm fine with either, I guess the default build process should be `npm install` then `npm start` and we can leave docker out of it.

**Pull request closed without merging**